### ISSUE TITLE
Bots calling member.getPresence() on themselves crashes D4J

### ISSUE THREAD

**Issue by LatvianModder** [Discord4J/Discord4J#838]
**To Reproduce:** 

```java
Member member = guild.getMemberById(client.getSelfId()).block();
Presence presence = member.getPresence().block();
Activity activity = presence.getActivity().orElse(null);
ReactionEmoji emoji = activity.getEmoji().orElse(null);
String state = activity.getState().orElse("");
System.out.println(emoji + " / " + state); // Doesn't even get this far
```

**Expected Behavior:** 
Parse the received json correctly and throw catchable error if something goes wrong

**Actual Behavior:** 
You get this error in console:

```
[reactor-http-epoll-2] WARN discord4j.gateway.payload.JacksonPayloadReader - Error while decoding JSON (com.fasterxml.jackson.databind.exc.MismatchedInputException: Cannot deserialize instance of `discord4j.discordjson.json.gateway.ImmutablePresenceUpdate$Json` out of START_ARRAY token
 at [Source: UNKNOWN; line: -1, column: -1] (through reference chain: discord4j.discordjson.json.gateway.ImmutableGuildMembersChunk$Json["presences"])): {"t":"GUILD_MEMBERS_CHUNK","s":536,"op":0,"d":{"presences":[{"user":{"id":"712800443566260316"},"status":"online","game":{"type":5,"name":"gnomebot.dev","id":"ec0b28a579ecb4bd","created_at":1611064029135},"client_status":{"web":"online"},"activities":[{"type":5,"name":"gnomebot.dev","id":"ec0b28a579ecb4bd","created_at":1611064029135}]}],"not_found":[],"nonce":"3628016705201799","members":[{"user":{"username":"Gnome","id":"712800443566260316","discriminator":"8531","bot":true,"avatar":"937bb0153de3e6c6de9efb01a3669c9d"},"roles":["214486855918288896","220243391936462849","765995096642158602","797618934350086155","763489207716675597","748075791790637078","721862495379259483"],"premium_since":null,"pending":false,"nick":null,"mute":false,"joined_at":"2021-01-10T00:12:53.046928+00:00","is_pending":false,"hoisted_role":"214486855918288896","deaf":false}],"guild_id":"166630061217153024","chunk_index":0,"chunk_count":1}}
```

**Version:** 
3.1.4-GH.818.0-SNAPSHOT

**Other:** 
Related PR (but didn't fix it entirely): https://github.com/Discord4J/Discord4J/pull/813

**Comment by quanticc** [comment]
Thanks for reporting it. This is a regression after #813 but before it you wouldn't get self presence information.

https://github.com/Discord4J/discord-json/blob/master/src/main/java/discord4j/discordjson/json/gateway/GuildMembersChunk.java needs to be updated to reflect that `presences` field is a `Possible<List<PresenceUpdate>>` ([reference](https://discord.com/developers/docs/topics/gateway#guild-members-chunk)) and then the appropriate code on D4J core when extracting its contents.

This "error" is not forwarded to you there because of how gateway works internally: core waits for a matching chunk and from its perspective it never arrives, and introducing an error would terminate the socket. Maybe a timeout can be applied for you.

**Comment by quanticc** [comment]
Fixed in latest `3.1.4-GH.818.0-SNAPSHOT` and `3.1.4-SNAPSHOT`
